contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -5487,4 +5491,37 @@ default_collation_for_builtin_type(Type typ, bool handle_pg_type)
 	}
 
 	return oid;
+}
+/*
+ * Special handling to resolve crash in some tests due to memory leak when running JDBC tests with PgAudit enabled. 
+ * When PgAudit is enabled we reach to ENRgetSystableScan() from postgres dialect which results that we are not 
+ * initialising the cache due to such handling in ENRgetSystableScan()
+ * Here we are changing the dialect for the table objects and then initialise the cache and abort it.
+ */
+static char*
+pltsql_get_object_identity_event_trigger(ObjectAddress* address)
+{
+    char *identity = NULL;
+    if(getObjectClass(address) == OCLASS_CLASS)
+    {
+        int save_nestlevel = 0;
```

Comment:
Question: Should this be volatile? or we can let the variable stay in stack?

Reply:
We don't want it to be volatile as if it's volatile it could get reassigned somewhere else which we can't easily track, can break other testfiles.